core/lib/types/src/l1_batch_committer.rs

```diff
@@ -0,0 +1,134 @@
+use zksync_basic_types::{ethabi::Token, U256};
+
+use crate::{commitment::L1BatchWithMetadata, utils};
+
+pub trait L1BatchCommitter
+where
+    Self: std::fmt::Debug + Send + Sync,
+{
+    fn l1_commit_data(&self, l1_batch_with_metadata: &L1BatchWithMetadata) -> Token;
+    fn l1_commit_data_size(&self, l1_batch_with_metadata: &L1BatchWithMetadata) -> usize {
+        crate::ethabi::encode(&[Token::Array(vec![
+            self.l1_commit_data(l1_batch_with_metadata)
+        ])])
+        .len()
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct RollupModeL1BatchCommitter {}
+
+#[derive(Debug, Clone)]
+pub struct ValidiumModeL1BatchCommitter {}
```

Comment:
This will eventually need to be migrated to the Component model in @popzxc's changes. @popzxc, please validate - do I get it right that this definition is fine for now?

Reply:
Yes, it does resemble a resource in the new framework notation, so it's OK.

Reply:
Eventually, this will be the case, let us know when that moment comes.